xdsl/transforms/DeadCodeElimination.py

```diff
@@ -0,0 +1,22 @@
+from xdsl.traits import Pure
+from xdsl.pattern_rewriter import (PatternRewriter, PatternRewriteWalker,
+                                   RewritePattern, GreedyRewritePatternApplier)
+from xdsl.ir import MLContext, Operation
+from xdsl.dialects.builtin import ModuleOp
+
+
+class UnusedOperationRemover(RewritePattern):
+
+    def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter, /):
+        if isinstance(op, Pure) and all(
+            [len(res.uses) == 0 for res in op.results]):
+            rewriter.erase_matched_op()
+
+
+def dce(ctx: MLContext, module: ModuleOp):
```

Comment:
Would it be possible to have a more descriptive name here? For me at least DCE is not a well-known abbreviation (although I have no real background in compilers, but I think our goal is to have more people without a formal background in compilers working with the project, right?)